src/main/java/io/confluent/connect/aws/cloudwatch/AwsCloudWatchSourceTask.java

```diff
@@ -19,32 +29,89 @@
     for logging
   */
 
+  public static String TASK_AWS_CLOUDWATCH_LOG_STREAM_NAME = "task.aws.cloudwatch.log.stream.name";
+  public static String AWS_CLOUDWATCH_PARTITION_KEY = "aws.cloudwatch.partition.key";
+  public static String AWS_CLOUDWATCH_OFFSET = "aws.cloudwatch.offset";
+
+  private AWSLogs client;
   static final Logger log = LoggerFactory.getLogger(AwsCloudWatchSourceTask.class);
   AwsCloudWatchSourceConnectorConfig config;
 
+  String logGroupName;
+  String taskLogStreamName;
+  String nextToken;
+  Map<String, String> sourcePartition;
+
   @Override
   public String version() {
     return Version.forClass(this.getClass());
   }
 
   @Override
   public void start(Map<String, String> settings) {
+
     config = new AwsCloudWatchSourceConnectorConfig(settings);
+    logGroupName = config.getString(AwsCloudWatchSourceConnectorConfig.AWS_CLOUDWATCH_LOG_GROUP_NAME);
+    taskLogStreamName = config.getString(TASK_AWS_CLOUDWATCH_LOG_STREAM_NAME);
+    nextToken = null;
+
+    sourcePartition = ImmutableMap.of(AWS_CLOUDWATCH_PARTITION_KEY, taskLogStreamName);
+
+    Map<String, Object> lastOffset = this.context.offsetStorageReader()
+            .offset(this.sourcePartition);
+
+    if (lastOffset != null && !lastOffset.isEmpty()) {
+      nextToken = (String) lastOffset.get(AWS_CLOUDWATCH_OFFSET);
+    }
+
+    client = buildAwsLogsClient(config);
+  }
 
-    //TODO: Do things here that are required to start your task.
-    //This could be open a connection to a database, etc.
+  private AWSLogs buildAwsLogsClient(AwsCloudWatchSourceConnectorConfig config) {
+    AWSLogsClientBuilder builder = AWSLogsClientBuilder.standard();
+    builder.setCredentials(config.getCredentialsProvider());
+    builder.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(config.getString(
+            AwsCloudWatchSourceConnectorConfig.AWS_CLOUDWATCH_URL_CONFIG), null));
+    return builder.build();
   }
 
   @Override
   public List<SourceRecord> poll() throws InterruptedException {
-    //TODO: Create SourceRecord objects that will be sent the kafka cluster.
-    // Throw ConnectException if there's a failure
-    // Throw RetriableException if Connect should try again
-    return Collections.emptyList();
+    List<SourceRecord> records;
+
+    GetLogEventsRequest request = new GetLogEventsRequest(logGroupName, taskLogStreamName);
+    request.setNextToken(nextToken);
+
+    GetLogEventsResult logEventsResult = client.getLogEvents(request);
+
+    String oldToken = nextToken;
+    nextToken = logEventsResult.getNextForwardToken();
+
+    if (nextToken.equals(oldToken)) {
+      return null;
+    }
+
+    List<OutputLogEvent> logEvents = logEventsResult.getEvents();
+    records = new ArrayList<>(logEvents.size());
```

Comment:
nit: declare it here
```suggestion
    List<SourceRecord> records = new ArrayList<>(logEvents.size());
```